src/Config.php

```diff
@@ -1045,4 +1044,34 @@ public function getCurrentTimestamp()
 
         return $now;
     }
+
+    protected function replaceRecursive($array1, $array2)
+    {
+        $merged = $array1;
+
+        foreach ($array2 as $key => &$value) {
+            // if $key = 'accept_file_types, don't merge..
+            if ($key == 'accept_file_types') {
+                $merged[$key] = $array2[$key];
+                continue;
+            }
+
+            if (isset($merged[$key]) && $merged[$key] instanceof Type\ResolvableInterface) {
+                $merged[$key]->update($value);
+            } else if (is_array($value) && isset($merged[$key]) && is_array($merged[$key])) {
+                $merged[$key] = $this->replaceRecursive($merged[$key], $value);
+            } else {
+                $merged[$key] = $value;
+            }
+        }
+
+        return $merged;
+    }
+
+    /**
+     * @deprecated Will be removed in Bolt 3.0
```

Comment:
Well we need this one to handle Resolvable objects and accept_file_types key. But all the other cases of Lib::mergeRecursiveDistinct can be replaced with array_replace_recursive.